UI/window-dock-youtube-app.cpp

```diff
@@ -0,0 +1,386 @@
+#include <sstream>
+#include <QUuid>
+
+#include "window-basic-main.hpp"
+#include "youtube-api-wrappers.hpp"
+#include "window-dock-youtube-app.hpp"
+
+#include "ui-config.h"
+
+#ifndef YT_WEBAPP_PLACEHOLDER
+#define YT_WEBAPP_PLACEHOLDER \
+	"https://studio.youtube.com/live/channel/UC/console?kc=OBS"
+#endif
+#ifndef YT_WEBAPP_ADDRESS
+#define YT_WEBAPP_ADDRESS \
+	"https://studio.youtube.com/live/channel/%1/console?kc=OBS"
+#endif
+
+#define BROADCAST_CREATED "BROADCAST_CREATED"
+#define BROADCAST_SELECTED "BROADCAST_SELECTED"
+#define INGESTION_STARTED "INGESTION_STARTED"
+#define INGESTION_STOPPED "INGESTION_STOPPED"
+#define LOGGED_OUT "LOGGED_OUT"
```

Comment:
These constants made as "#define"s to be consisted with other youtube-related code. Like `youtube-api-wrappers.cpp`, for example. I think we can keep this as defines?

Reply:
Whether they are consistent with other existing code is not relevant to the question at hand. We do not assume existing patterns to be "good" patterns.

Reply:
Correct - if there is a language construct for the purpose, it is to be used and always preferred over macro definitions. We are in the process of cleaning this up in our codebase step by step and expect new code to avoid the same old patterns.

Static strings should always be set up as character array constants and should not take the value of preprocessor definitions (which in this case would make the checks for `YT_WEBAPP_ADDRESS` superfluous as it should be changeable in the first place).

Reply:
Done.